### PR TITLE
Feature/network

### PR DESCRIPTION
**Pull request by TStrawberry**
## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## In scope
- Add Network root implementation
- Add interceptor

## Check list
1. [x] Does your submission pass tests?
2. [x] Have you lint your code locally prior to submission?
3. [x] Have you update the wiki of your change?

## Reference
- Card: https://trello.com/c/wlINmCUX

### CODE REVIEW

File: modules/Core/Sources/Network/DataSession.swift [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        case unreachableNetwork
+        case unknown(description: String?)
+    }
+
+    public typealias Request = URLRequest
+    public typealias Response = Data
+
+    let session: Alamofire.Session
+
+    var isNetworkReachable: () -> Bool = { NetworkReachabilityManager.default?.isReachable ?? false }
```
Comment (Tony---Zhang) [inline modules/Core/Sources/Network/DataSession.swift]: 可以把这些封装到不同的拦截器中吗？
比如这个可以叫“NetworkReachabilityInterceptor”

然后这些Interceptor都集成一个标准的Interceptor协议

protocol Interceptor {
fun intercept(request: Request) -> ??
}

File: modules/Core/Sources/Network/DataSession.swift [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+    public init(session: Alamofire.Session = .default) {
+        self.session = session
+    }
+
+    public func resume(_ request: URLRequest, response: @escaping (Result<Data, Swift.Error>) -> Void) {
```
Comment (Tony---Zhang) [inline modules/Core/Sources/Network/DataSession.swift]: 这个写法，我们需要修改成combine的写法，而不用写成callback这种

File: modules/Core/Sources/Network/Session.swift [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+extension Session {
+    public func intercept(_ request: Request, completion: @escaping (Result<Request, Error>) -> Void) {
+        completion(Result.success(request))
+    }
+
+    public func intercept(_ response: Response, completion: @escaping (Result<Response, Error>) -> Void) {
+        completion(Result.success(response))
+    }
+
+    public func resumeWithInterceptor(_ request: Request, response: @escaping (Result<Response, Error>) -> Void) {
```
Comment (Tony---Zhang) [inline modules/Core/Sources/Network/Session.swift]: 这个方法不要暴露出去吧？我们需要基于接口编程。我理解所有的接口都应该经过我们设定的拦截器

**Review by Tony---Zhang — CHANGES_REQUESTED** [review]
Need to change to combine style

func fetch(_ request: T) -> AnyPublisher<Result<Data, Swift.Error>>

**Review by pujiaxin33 — APPROVED** [review]

**Review by Tony---Zhang — APPROVED** [review]

**Pull request merged**